File: py/IEC61968/Customers/ServiceCategory.py
from CIM_STD_PYTHON.TC57CIM.IEC61968.Common.ConfigurationEvent import ConfigurationEvent
from CIM_STD_PYTHON.TC57CIM.IEC61968.Customers.PricingStructure import PricingStructure
from CIM_STD_PYTHON.TC57CIM.IEC61968.Customers.ServiceKind import ServiceKind
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.IdentifiedObject import IdentifiedObject


###########################################################
#  service_category.py
#  Implementation of the Class ServiceCategory
#  Created on:      19-Dec-2023 3:28:49 PM
###########################################################


class ServiceCategory(IdentifiedObject):

    def __init__(self):
        """
        Constructor for ServiceCategory class.
        """
        super().__init__()
        self.kind = ServiceKind.ELECTRICITY
        self.pricing_structures = PricingStructure()
        self.configuration_events = ConfigurationEvent()

File: java/IEC61968/Customers/ServiceCategory.java
package IEC61968.Customers;

import IEC61970.Base.Core.IdentifiedObject;
import IEC61968.Common.ConfigurationEvent;

/**
 * Category of service provided to the customer.
 * @created 03-Jan-2024 12:17:44 PM
 */
public class ServiceCategory extends IdentifiedObject {

	/**
	 * Kind of service.
	 */
	public ServiceKind kind;
	/**
	 * All pricing structures applicable to this service category.
	 */
	public PricingStructure PricingStructures;
	/**
	 * All configuration events created for this service category.
	 */
	public ConfigurationEvent ConfigurationEvents;

	public ServiceCategory(){

	}
}//end ServiceCategory